src/zenml/models/v2/core/artifact_visualization.py

```diff
@@ -105,6 +117,23 @@ def artifact_version_id(self) -> UUID:
         """
         return self.get_metadata().artifact_version_id
 
+    @property
+    def artifact_version(self) -> "ArtifactVersionResponse":
+        """The artifact version resource, if the response was hydrated with it.
+
+        Returns:
+            The artifact version resource associated with this visualization.
+
+        Raises:
+            RuntimeError: If the artifact version resource is not available.
+        """
+        resources = self.get_resources()
```

Comment:
`self.get_resources()` will always hydrate the response, so you can remove all the code below (See any property of any other model that returns something from `resources`)